src/test/java/ArrayIndexedBagTest.java

```diff
@@ -342,4 +343,107 @@ void removeByNegativeIndexOfIndexedBagPropagatesException() {
         IndexedBag<Integer> bag = new ArrayIndexedBag<>();
         assertThrows(IndexOutOfBoundsException.class, () -> bag.remove(-1));
     }
+
+    @Test
+    void iteratorReturnsIterator() {
+        IndexedBag<Integer> bag = new ArrayIndexedBag<>();
+        Iterator<Integer> it = bag.iterator();
+        assertNotNull(it);
+        assertTrue(it instanceof Iterator);
+    }
```

Comment:
Remove, not really testing anything. It's always going to be an `Iterator` (that's the signature) and non-null-ness will be implicit from other tests.